### PR TITLE
OSSM-3266: Update maistra prowconfig

### PR DESCRIPTION
**Pull request by yxun**

### CODE REVIEW

File: core-services/prow/02_config/maistra/_prowconfig.yaml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 branch-protection:
   orgs:
     maistra:
+      protect: true
       unmanaged: true
```
Comment (jwendell) [inline core-services/prow/02_config/maistra/_prowconfig.yaml:5]: I have no idea what those settings mean. Can they coexist?

Comment (yxun) [reply core-services/prow/02_config/maistra/_prowconfig.yaml:5]: well, I can remove the "protect:" one. The branch-protection has been in maistra repos. Keep "unmanaged" can avoid overwriting the existing github branch config.

Comment (jwendell) [reply core-services/prow/02_config/maistra/_prowconfig.yaml:5]: sounds good.

**Review by jwendell — APPROVED** [review]

**Pull request merged**